### PR TITLE
feat: Add common Messaging bootstrap handler

### PR DESCRIPTION
**Pull request by lenny-intel**
This is needed as other core/support service start using the MessageBus
for Service Metrics and System Events.

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A for internal code**
  <link to docs PR>

## Testing Instructions
Use this branch in edgex-go
Modify core -data to:
- Use new handler from this branch
- Add setting MessageQueue in result returned by GetBootstrap()
- Build an run Core Data.
- Verify runs fine and receives data from the MessageBus

### CODE REVIEW

File: bootstrap/handlers/messaging.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	if len(messageQueue.Host) == 0 {
+		lc.Error("MessageQueue configuration not set or missing from service's GetBootstrap() implementation")
+		return false
+	}
+
+	// Make sure the MessageBus password is not leaked into the Service Config that can be retrieved via the /config endpoint
+	messageBusInfo := deepCopy(messageQueue)
+
+	messageBusInfo.AuthMode = strings.ToLower(strings.TrimSpace(messageBusInfo.AuthMode))
+	if len(messageBusInfo.AuthMode) > 0 && messageBusInfo.AuthMode != boostrapMessaging.AuthModeNone {
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply bootstrap/handlers/messaging.go:48]: I like it! Thanks for pointing this out.

File: bootstrap/handlers/messaging.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		if err := boostrapMessaging.SetOptionsAuthData(&messageBusInfo, lc, dic); err != nil {
+			lc.Error(err.Error())
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply bootstrap/handlers/messaging.go:50]: Done

File: bootstrap/handlers/messaging.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lenny-intel) [reply bootstrap/handlers/messaging.go]: reworked

**Comment by lenny-intel** [comment]
@cloudxxx8 , if you approve PR, please merge.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**